pkg/operator/operator.go

```diff
@@ -275,6 +279,14 @@ func (c *Controller) enqueueUpdateJob(old, new interface{}) {
 		return
 	}
 
+	// if there is a change in the chartsource (ref change, eg),
+	// its possible that the mirror's ref-sha be obsolete w.r.t. upstream
+	// repo's ref-sha. To avoid spurious deploy, mirror is synced before
+	// doing helm reconciliation
+	if csDiff := cmp.Diff(oldHr.Spec.ChartSource,newHr.Spec.ChartSource); csDiff != "" {
+		c.apiServer.SyncMirrors()
```

Comment:
I do not see how this is different from what the controller already does, as during the preparation stage [`GetMirrorCopy` is called](https://github.com/fluxcd/helm-operator/blob/90f49460e252bb60b3c8fe5593854c2c716c2236/pkg/release/release.go#L143), which automatically results in an additional sync: https://github.com/fluxcd/helm-operator/blob/90f49460e252bb60b3c8fe5593854c2c716c2236/pkg/chartsync/git.go#L169

Reply:
IIUC, this sync ^ is between helmrelease sourceref and helm. It assumes Mirror is uptodate and works to reconcile helmrelease. 
while ```c.apiServer.SyncMirrors()``` <- this one is basically `git fetch --tags origin` for each Mirror.
Please let me know if I am wrong. I tested this behaviour locally, before creating a PR. thanks.

Reply:
I think you are indeed correct, what worries me about using `SyncMirrors` it that it triggers it for each mirror, while the task is just for a single repository. This is depending on the amount of repositories it mirrors, either inefficient or _really_ inefficient.

Reply:
yes, it would be inefficient. For us, we are only using single repo. 
Before doing that optimization, wanted a goahead from the project authors.
I can try my hand at it.

Reply:
You have my blessings to go ahead and change it, but be aware that the efforts will likely have a short lifetime, as we are slowly moving users to the [helm-controller](https://github.com/fluxcd/helm-controller).

Reply:
yes, thanks. aware of the short lifetime. this fix is important to us since we are using it in production and its impacting our operations. we want to fix it before migrating to v2.